Base/QTCore/qSlicerCoreApplication.cxx

```diff
@@ -1720,19 +1720,47 @@ void qSlicerCoreApplication::onAboutToQuit()
 #endif
 }
 
+//----------------------------------------------------------------------------
+inline QStringList findTranslationFilesWithLanguageExtension(const QString& dir, const QString& languageExtension)
+{
+  const QString localeFilter = QString("*%1.qm").arg(languageExtension);
+  return QDir(dir).entryList(QStringList(localeFilter));
+}
+
+//----------------------------------------------------------------------------
+inline QStringList findTranslationFiles(const QString& dir, const QString& settingsLanguage)
+{
+  // If settings language is empty don't search translation files (application default)
+  if (settingsLanguage.isEmpty())
+    {
+    return {};
+    }
+
+  // Try to find the translation files using specific language extension
+  // (In case of fr_FR -> look for *fr_FR.qm files).
+  auto files = findTranslationFilesWithLanguageExtension(dir, settingsLanguage);
```

Comment:
I like `auto` for iterator variables and other places where there'd be a spew of repeated template type noise, but I agree that in some cases like this spelling out the type would make the code easier to comprehend.

Reply:
I completely agree. Using auto makes the code harder to read. I have experienced the same thing that @lassoan describes when reading C++ code. Of course in Python this is commonplace and cannot be prevented, but it would make sense to keep more explicitness in C++. I also agree with @pieper's comment about iterator variables, at those places the type is obvious.